### PR TITLE
Audioplayer attachment

### PR DESCRIPTION
**Pull request by JeanTracker**
## AudioPlayerAgent: Add TTSPlayer

- Implement the attachment spec for AudioPlayerAgent.

### CODE REVIEW

**Comment by webispy** [comment]
How about split this PR to 2 PRs ? Because too large(difficult) to review the PR.

PR-1:  PCM / Portaudio related PR
- Pcm: Add get_position and get_duration
- portaudio: Add debug for playback
- Pcm: fix the heap-use-after-free
- Portaudio: fix the double free heap

And,

PR-2: TTSPlayer
- Add TTSPlayer
- Refactoring TTSAgent
- AudioPlayerAgent: add TTSPlayer

**Comment by JeanTracker** [comment]
> How about split this PR to 2 PRs ? Because too large(difficult) to review the PR.
> 
> PR-1: PCM / Portaudio related PR
> 
> * Pcm: Add get_position and get_duration
> * portaudio: Add debug for playback
> * Pcm: fix the heap-use-after-free
> * Portaudio: fix the double free heap
> 
> And,
> 
> PR-2: TTSPlayer
> 
> * Add TTSPlayer
> * Refactoring TTSAgent
> * AudioPlayerAgent: add TTSPlayer

Thanks. I'll split this PR.

File: src/capability/audio_player_agent.cc [hunk 1]
```diff
@@ -91,20 +103,68 @@ void AudioPlayerAgent::restore()
...
+    if (is_paused) {
+        nugu_warn("AudioPlayer is pause mode caused by directive(PAUSE)");
         return;
+    }
 
-    if (!player->play()) {
-        nugu_error("play media failed");
-        sendEventPlaybackFailed(PlaybackError::MEDIA_ERROR_INTERNAL_DEVICE_ERROR,
-            "player can't play");
+    IMediaPlayer* player = is_tts_activate ? tts_player : media_player;
```
Comment (kimhyungrok) [inline src/capability/audio_player_agent.cc]: This is so many exists. I think that it's possible to define IMediaPlayer* player to class member variable and set this value in parsingPlay() one times.

Comment (JeanTracker) [reply src/capability/audio_player_agent.cc]: Thanks. I'll check this.

**Pull request merged**